test/test_biogeochemistry.jl

```diff
@@ -12,6 +12,26 @@ import Oceananigans.Biogeochemistry:
     biogeochemical_advection_scheme, 
     update_biogeochemical_state!
 
+@kernel function update_PAR!(PAR, grid, P, t, bgc) 
+    i, j = @index(Global, NTuple) 
+    
+    surface_PAR = bgc.surface_PAR(t)
+    
+    z = grid.zᵃᵃᶜ[grid.Nz]
+    
+    ∫chl = - P[i, j, grid.Nz]^bgc.phytoplankton_light_attenuation_exponent*z
+    PAR[i, j, grid.Nz] =  surface_PAR*exp(bgc.water_light_attenuation_coefficient * z - bgc.phytoplankton_light_attenuation_coefficient * ∫chl)
+    for k=grid.Nz-1:-1:1
+        z = grid.zᵃᵃᶜ[k]
+        dz = grid.zᵃᵃᶜ[k+1] - z 
+    
+        ∫chl += P[i, j, grid.Nz]^bgc.phytoplankton_light_attenuation_exponent*dz
+    
+        PAR[i, j, k] =  PAR[i, j, k+1]*exp(- bgc.water_light_attenuation_coefficient * dz - bgc.phytoplankton_light_attenuation_coefficient * ∫chl)
+    end
+end 
```

Comment:
Take care with style issues / inconsistent whitespace above

Reply:
It also may help to unpack some of the parameters, eg

```julia
n = bgc.phytoplankton_light_attenuation_exponent
```

or whatever concise mathematical notation is used.

Reply:
Also this won't work on the GPU.